### PR TITLE
winafl-cmin: Refactor so that main isn't 350+ LOC

### PR DESCRIPTION
**Pull request by moshekaplan**

### CODE REVIEW

**Comment by moshekaplan** [comment]
Can you double-check that the checks with the non-default `--working-dir` have the same behavior as before? That's the only part that I'm not confident about.

File: winafl-cmin.py [hunk 1]
```diff
@@ -294,134 +296,119 @@ def setup_argparse():
...
-    logging.debug(
-        'Inspecting the following directories: %s',
-        ', '.join(args.input)
-    )
-    inputs = []
-    for path in args.input:
-        for root, dirs, files in os.walk(path):
-            for file_ in files:
-                inputs.append(os.path.join(root, file_))
+        if not os.path.isdir(dir_path):
```
Comment (ifratric) [inline winafl-cmin.py:402]: I assume there should be an error message printed for this case?

**Review by ifratric — COMMENTED** [review]
WRT the working_dir, I see the change there is to also treat args.file_read and args.input as explicitly relative to the working_dir (this is essentially a NOP since the current directory gets changed to working_dir anyway, so any relative path is going to be relative to that). It's questionable, in the case that working_dir is specified and args.input etc. is relative, the user meant relative to the working_dir or relative to the current directory when the command was run, but your PR keeps the current behavior so it seems fine.

I have one other small comment (see inline).

File: winafl-cmin.py (hunk 1 above)
Comment (moshekaplan) [reply winafl-cmin.py:402]: ack, good idea. It was a replacement for the `assert all(os.path.isdir(i) for i in args.input)`, which previously didn't have an error message.

**Comment by ifratric** [comment]
Thanks!

**Pull request merged**